pkg/services/rediscache/provision.go

```diff
@@ -22,67 +20,64 @@ func (s *serviceManager) GetProvisioner(
 func (s *serviceManager) preProvision(
 	context.Context,
 	service.Instance,
-) (service.InstanceDetails, service.SecureInstanceDetails, error) {
-	dt := instanceDetails{
+) (service.InstanceDetails, error) {
+	dt := &instanceDetails{
 		ARMDeploymentName: uuid.NewV4().String(),
 		ServerName:        uuid.NewV4().String(),
 	}
-	dtMap, err := service.GetMapFromStruct(dt)
-	return dtMap, nil, err
+
+	return dt, nil
```

Comment:
Nit: Let's return a struct literal here instead of assigning to `dt`.

Reply:
Updated it.